File: js/그림(1926).js
const input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');

const [n, m] = input[0].split(' ').map(Number);
const graph = [];
const visited = Array.from({length: n}, () => Array(m).fill(false));

for (let i=1; i<=n; i++) {
    const row = input[i].split(' ').map(Number);
    graph.push(row);
}

const dx = [-1,1,0,0];
const dy = [0,0,-1,1];
const extent_arr = [];

function bfs(x,y) {
    const queue = [];
    queue.push([x,y]);
    visited[x][y] = true;
    let extent = 1;

    while (queue.length) {
        const [x,y] = queue.shift();

        for (let i=0; i<4; i++) {
            const nx = x + dx[i];
            const ny = y + dy[i];

            if (nx >= 0 && nx < n && ny >= 0 && ny < m && !visited[nx][ny] && graph[nx][ny] === 1) {
                visited[nx][ny] = true;
                graph[nx][ny] = graph[x][y] + 1;
                queue.push([nx,ny]);
                extent++;
            }
        }
    }
    extent_arr.push(extent);
}

let cnt = 0;
for (let i=0; i<n; i++) {
    for (let j=0; j<m; j++) {
        if (graph[i][j] === 1) {
            bfs(i,j);
            cnt++;
        }
    }
}

if (extent_arr.length > 0) {
    console.log(cnt);
    console.log(Math.max(...extent_arr));
} else {
    console.log(0);
    console.log(0);
}

File: python/그림(1926).py
from collections import deque

n, m = tuple(map(int, input().split()))
arr = [list(map(int, input().split())) for _ in range(n)]
visited = [[False] * m for _ in range(n)]

dx = [-1,1,0,0]
dy = [0,0,-1,1]

extent_arr = []

def bfs(x,y):
    queue = deque()
    queue.append((x,y))
    visited[x][y] = True
    global extent
    extent = 1

    while queue:
        x, y = queue.popleft()

        for i in range(4):
            nx = x +dx[i]
            ny = y +dy[i]
            
            if nx >= 0 and nx < n and ny >= 0 and ny < m and arr[nx][ny] == 1 and visited[nx][ny] == False:
                visited[nx][ny] = True
                arr[nx][ny] = arr[x][y] + 1
                extent += 1
                queue.append((nx,ny))
    extent_arr.append(extent)
    
cnt = 0
for i in range(n):
    for j in range(m):
        if arr[i][j] == 1:
            bfs(i,j)
            cnt += 1

if extent_arr:
    print(cnt)
    print(max(extent_arr))
else:
    print(0)
    print(0)

# [[1, 2, 0, 1, 2],
#  [0, 3, 4, 0, 0],
#  [0, 0, 0, 0, 0],
#  [1, 0, 1, 2, 3],
#  [0, 0, 2, 3, 4],
#  [0, 0, 3, 4, 5]] 4